tests/transport_test.go

```diff
@@ -0,0 +1,118 @@
+package tests
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"kubevirt.io/containerized-data-importer/pkg/common"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	"k8s.io/api/core/v1"
+)
+
+var _ = Describe("Transport Tests", func() {
+
+	const (
+		annEndpoint = "cdi.kubevirt.io/storage.import.endpoint"
+		annSecret   = "cdi.kubevirt.io/storage.import.secretName"
+
+		fileHostName   = "cdi-file-host" // deployment and service name
+		fileHostNs     = "kube-system"   // deployment and service namespace
+		fileHostUname  = "admin"         // http and s3 username
+		fileHostPasswd = "password"      // http and s3 password
+
+		httpPortName = "http"
+		s3PortName   = "s3"
+
+		targetFile = "tinyCore.iso"
+		bucket     = "images"
+	)
+
+	var (
+		httpEp, s3Ep, ns string
+		fileHostService  *v1.Service // persistent file host service, routes to http and s3 ports
+	)
+
+	f, err := framework.NewFramework("", framework.Config{false})
+	if err != nil {
+		Fail(errors.Wrap(err, "Failed to create test framework").Error())
+	}
+
+	c, err := f.GetKubeClient()
+	if err != nil {
+		Fail(errors.Wrap(err, "Failed to create k8s client").Error())
+	}
+
+	cdiLabel := map[string]string{common.CDI_LABEL_KEY: common.CDI_LABEL_VALUE}
+
+	BeforeEach(func() {
+		ns = f.Namespace.Name
+
+		By(fmt.Sprintf("Waiting for all \"%s/%s\" deployment replicas to be Ready", fileHostNs, fileHostName))
+		utils.WaitForDeploymentReplicasReadyOrDie(c, fileHostNs, fileHostName)
+
+		By(fmt.Sprintf("Getting file host service \"%s/%s\"", fileHostNs, fileHostName))
+		fileHostService = utils.GetServiceInNamespaceOrDir(c, fileHostNs, fileHostName)
+		Expect(fileHostService).NotTo(BeNil())
+
+		By("Generating s3 and http endpoints urls")
+		clusterIP := fileHostService.Spec.ClusterIP
+		Expect(clusterIP).NotTo(BeEmpty(), "Failed to get clusterIP from fileHostService")
+
+		httpPort := utils.GetServicePortByName(fileHostService, httpPortName)
+		Expect(httpPort).NotTo(BeZero(), "http port is 0")
+
+		s3Port := utils.GetServicePortByName(fileHostService, s3PortName)
+		Expect(s3Port).NotTo(BeZero(), "s3 port is 0")
+
+		httpEp = fmt.Sprintf("http://%s:%d/%s", clusterIP, httpPort, targetFile)
+		s3Ep = fmt.Sprintf("http://%s:%d/%s/%s", clusterIP, s3Port, bucket, targetFile)
+	})
+
+	tests := []struct {
+		title       string
+		urlEndpoint string
+		secretName  string
+	}{
+		{
+			"should connect to http endpoint without credentials",
+			httpEp,
+			"",
+		},
+		{
+			"should connect to http endpoint with credentials",
+			httpEp,
+			"",
+		},
+		{
+			"should connect to s3 endpoint without credentials",
+			s3Ep,
+			"",
+		},
+		{
+			"should connect to s3 endpoint with credentials",
+			s3Ep,
+			"",
+		},
+	}
```

Comment:
any value in an _https_ test case?

Reply:
Go's http client handles http(s) seamlessly, meaning it's not a factor in the code we've written.  To me that means we'd be testing go source code and not the importer.